common/src/main/scala/explore/cache/ProgramCache.scala

```diff
@@ -43,13 +43,47 @@ case class ProgramCache(
 object ProgramCache extends CacheComponent[ProgramSummaries, ProgramCache]:
   given Reusability[ProgramCache] = Reusability.by(_.programId)
 
+  private def drain[A, Id, R](
+    fetch:      Option[Id] => IO[R],
+    getList:    R => List[A],
+    getHasMore: R => Boolean,
+    getId:      A => Id
+  ): IO[List[A]] = {
+    def go(id: Option[Id], accum: List[A]): IO[List[A]] =
```

Comment:
tailrec?

Reply:
Is this fetching results in batches? cool

Reply:
> tailrec?

I tried that initially but it doesn't work because of the flatmap, which IIRC is stack safe for IO.